examples/permissioned-token/assets/permissions.py

```diff
@@ -0,0 +1,156 @@
+        Return(Int(1))
+    ])
+
+    # fetch permissions manager, token_id from controller's global state (using foreignApps)
+    permissions_manager = App.globalGetEx(Int(1), Bytes("manager"))
+    token_id = App.globalGetEx(Int(1), Bytes("token_id"))
+
+    """
+    Whitelist an account. If a non-reserve account is whitelisted then it can receive tokens.
+    Only accepted if txn sender is the permissions manager.
+    1 expected argument:
+    * controller_app_id: controller application index
+    NOTE: controller_app_id is also passed in txn.ForeignApps (to load perm_id, manager from controller's global state)
+    """
+    add_whitelist = Seq([
+        permissions_manager,
+        token_id,
+        Assert(And(
+    		Txn.application_args.length() == Int(2),
+    		Txn.accounts.length() == Int(1),
+    		rekey_check,
+    		# Txn.applications.length() ==  Int(1), [TEALv3]
+
+    		# verify from global state whether controller application passed is the valid one
+    		# note: with tealv3 we can just use txn.applications[0] instead of passing an an appArg
+    		Btoi(Txn.application_args[1]) == App.globalGet(controller_app_id),
+
+    		# first verify correct token value is passed
+    		# Txn.assets[0] == token_id.value(), [TEALv3]
+
+    		# then verify txn sender is the permissions manager
+    		Txn.sender() == permissions_manager.value(),
+        )),
+
+        If(
+            # If account is not already whitelisted, update the counter and continue, return otherwise
+            App.localGet(Int(1), Bytes("whitelisted")) == Int(0),
+            App.globalPut(whitelist_count, App.globalGet(whitelist_count) + Int(1)), # only evaluated if above is true
+            Return(Int(1)) # else return
+        ),
+
+        # finally update txn.accounts[1] local state to set whitelisted status as true(1)
+        App.localPut(Int(1), Bytes("whitelisted"), true),
+        Return(Int(1))
+    ])
+
+    asset_balance = AssetHolding.balance(Int(1), Gtxn[1].xfer_asset())
+
+    """
+    Transfer token from accA -> accB. Both A, B are non-reserve accounts.
+    Expected arguments:
+    * toAccountAddress (fetched from Txn.ForeignApps[0])
+    """
+    transfer_token = Seq([
+        asset_balance, # load asset_balance of asset_receiver from store
+        Assert(And(
+            Gtxn[1].type_enum() == TxnType.AssetTransfer, # this should be clawback call
+
+    		# verify tx.accounts[1] of current_tx should be same as asset receiver
+    		Txn.accounts[1] == Gtxn[1].asset_receiver(),
+
+    		# rule 1 - check balance of receiver after receiving token <= 100(max_tokens)
+    		asset_balance.value() <= App.globalGet(max_tokens),
+
+    		# rule 2 - [from, to] accounts must be whitelisted
+    		# NOTE: Int(0) == Txn.Sender(), Int(1) == Txn.accounts[1]
+    		App.localGet(Int(0), Bytes("whitelisted")) == true, # from account must be whitelisted
+    		App.localGet(Int(1), Bytes("whitelisted")) == true  # to account must be whitelisted
```

Comment:
wrong indentation - each level should be 4 spaces.

Reply:
It looks correct on my editor, and while viewing file. In diff it looks incorrect

Reply:
it was an issue with my editor

Reply:
It's good to use some extension in editor to mark tabs. For, example, I mark tabs with a low contrast `>` symbol: 

![image](https://user-images.githubusercontent.com/811701/117222902-d4534680-ae0c-11eb-8b20-5500d0b51c16.png)

Reply:
thanks for info!